numpydoc/tests/test_numpydoc.py

```diff
@@ -0,0 +1,55 @@
+# -*- encoding:utf-8 -*-
+from __future__ import division, absolute_import, print_function
+
+from numpydoc.numpydoc import mangle_docstrings
+from sphinx.ext.autodoc import ALL
+
+class MockConfig():
+    numpydoc_use_plots = False
+    numpydoc_use_blockquotes = True
+    numpydoc_show_class_members = True
+    numpydoc_show_inherited_class_members = True
+    numpydoc_class_members_toctree = True
+    templates_path = []
+    numpydoc_edit_link = False
+    numpydoc_citation_re = '[a-z0-9_.-]+'
+
+class MockBuilder():
+    config = MockConfig()
+
+class MockApp():
+    config = MockConfig()
+    builder = MockBuilder()
+    translator = None
+
+
+app = MockApp()
+app.builder.app = app
+
+def test_mangle_docstrings():
+    s ='''
+A top section before
+
+.. autoclass:: str
+    '''
+    lines = s.split('\n')
+    doc = mangle_docstrings(MockApp(), 'class', 'str', str, {}, lines)
+    assert 'rpartition' in [x.strip() for x in lines]
+
+    lines = s.split('\n')
+    doc = mangle_docstrings(MockApp(), 'class', 'str', str, {'members':['upper']}, lines)
+    assert 'rpartition' not in [x.strip() for x in lines]
+    assert 'upper' in [x.strip() for x in lines]
+    
+    lines = s.split('\n')
+    doc = mangle_docstrings(MockApp(), 'class', 'str', str, {'exclude-members': ALL}, lines)
+    assert 'rpartition' not in [x.strip() for x in lines]
+    assert 'upper' not in [x.strip() for x in lines]
+
+    lines = s.split('\n')
+    doc = mangle_docstrings(MockApp(), 'class', 'str', str, {'exclude-members': 'upper'}, lines)
```

Comment:
is upper here intentionally a string and not a list of strings? are both meant to be supported?

Reply:
Not really, but it works because `'upper' in 'upper' => True`. fixing